// internetShop/actions/actionTypes.js
export const FETCH_PHONES_REQUEST = 'FETCH_PHONES_REQUEST'
export const FETCH_PHONES_SUCCESS = 'FETCH_PHONES_SUCCESS'
export const FETCH_PHONES_FEILURE = 'FETCH_PHONES_FEILURE'

export const LOAD_MORE_PHONES_REQUEST = 'LOAD_MORE_PHONES_REQUEST'
export const LOAD_MORE_PHONES_SUCCESS = 'LOAD_MORE_PHONES_SUCCESS'
export const LOAD_MORE_PHONES_FEILURE = 'LOAD_MORE_PHONES_FEILURE'

export const FETCH_PHONE_BY_ID_REQUEST = 'FETCH_PHONE_BY_ID_REQUEST'
export const FETCH_PHONE_BY_ID_SUCCESS = 'FETCH_PHONE_BY_ID_SUCCESS'
export const FETCH_PHONE_BY_ID_FEILURE = 'FETCH_PHONE_BY_ID_FEILURE'

export const ADD_PHONE_TO_BASKET = 'ADD_PHONE_TO_BASKET'

export const SEARCH_PHONE = 'SEARCH_PHONE'

export const FETCH_CATEGORIES_REQUEST = 'FETCH_CATEGORIES_REQUEST'
export const FETCH_CATEGORIES_SUCCESS = 'FETCH_CATEGORIES_SUCCESS'
export const FETCH_CATEGORIES_FEILURE = 'FETCH_CATEGORIES_FEILURE'

export const CLEAR_BASKET = 'CLEAR_BASKET'

export const CHECKOUT_BASKET = 'CHECKOUT_BASKET'

export const REMOVE_PHONE_FROM_BASKET = 'REMOVE_PHONE_FROM_BASKET'

export const AUTH_REQUEST = 'AUTH_REQUEST'
export const AUTH_SUCCESS = 'AUTH_SUCCESS'
export const AUTH_FEILURE = 'AUTH_FEILURE'
