// containers/auth/useStyles.js
import {makeStyles} from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100vh',
    backgroundColor: '#63bdf1',
  },
  form: {
    background: '#fff',
    padding: 25,
    borderRadius: 5,
    display: 'flex',
    flexDirection: "column",
  },
  header: {
    marginBottom: 20,
    '& h3': {
      fontWeight: "bold",
    },
  },
  textField: {
    marginBottom: 10,
  },
  forgotPass: {
    alignSelf: "flex-end",
    color: "#0277bd",
    textDecoration: "none",
    fontSize: 12,
  },
  button: {
    marginTop: 30,
  },
  privacyPolicy: {
    alignSelf: "center",
    color: "#0277bd",
    textDecoration: "none",
    fontSize: 12,
    marginTop: 10,
  },
  cancel: {
    alignSelf: "flex-end",
    color: "#0277bd",
    textDecoration: "none",
    fontSize: 12,
    marginTop: 35,
  },
  buttonRecovery: {
    marginTop: 10,
  },
  headerRecovery: {
    marginBottom: 10,
    '& h3': {
      fontWeight: "bold",
    },
  },
});

export default useStyles
