// internetShop/containers/authMui/Recovery/index.js
import React, { useState } from 'react'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
// import Paper from '@material-ui/core/Paper'

import { validateControl } from 'internetShop/selectors'

const AuthMui = () => {
  const [isFormValid, setIsFormValid] = useState(false)
  const [formControls, setFormControls] = useState({
    email: {
      type: 'email',
      label: 'Email',
      value: '',
      valid: true,
      touched: false,
      helperText: '',
      validation: {
        required: true,
        email: true,
      },
    },
  })

  const handleChange = (e, controlName) => {
    const newFormControls = { ...formControls }
    const control = newFormControls[controlName]

    control.value = e.target.value
    control.touched = true
    control.valid = validateControl(e.target.value, control.validation)
    control.helperText = control.valid
      ? ''
      : 'Incorrect entry.'

    setFormControls(newFormControls)

    let isFormValid = true
    Object.keys(formControls).forEach(name => {
      isFormValid = formControls[name].valid && isFormValid
    })

    setIsFormValid(isFormValid)
  }

  const renderFields = () => {
    return (
      Object.keys(formControls).map((controlName, index) => {
        const control = formControls[controlName]

        return (
          <TextField
            key={index}
            id={`${control.label}-${index}`}
            type={control.type}
            value={control.value}
            label={control.label}
            helperText={control.helperText}
            error={!control.valid}
            variant='outlined'
            size='small'
            onChange={(e) => handleChange(e, controlName)}
          />
        )
      })
    )
  }

  const handleClick = e => {
    e.preventDefault()

    const newFormControls = { ...formControls }
    Object.keys(newFormControls).map(control => {
      if (isFormValid) newFormControls[control].value = ''
    })
    setFormControls(newFormControls)
  }

  return (
    <div className='auth'>
      <form className='auth__form'>
        <h3>Recovery Password</h3>
        {renderFields()}

        <div>
          <Button
            // href='/'
            variant='contained'
            color='primary'
            fullWidth={true}
            // disabled={!isFormValid}
            onClick={handleClick}
          >Recovery</Button>
        </div>
      </form>
    </div>
  )
}

export default AuthMui
