// containers/TopLine/topLineStyles.js
import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  root: {
    flexGrow: 1,
    minWidth: 320,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  list: {
    width: 250,
  },
  link: {
    textDecoration: "none",
    color: "#0277bd",
  },
  listIcon: {
    justifyContent: 'center',
  },
  listText: {
    '& span': {
      fontSize: "1.3rem",
    },
  },
  mobileLink: {
    textDecoration: "none",
    color: "#0277bd",
  },
  signUp: {
    fontSize: 16,
    color: "white",
    textDecoration: "none",
    '&:hover': {
      opacity: .7,
    },
  },
  signIn: {
    opacity: .7,
    '& MuiTypography-root:hover': {
      opacity: .7,
    },
  },
  sectionDesctop: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "flex",
      alignItems: "center",
    },
  },
  sectionMobile: {
    display: "flex",
    [theme.breakpoints.up("sm")]: {
      display: "none",
    },
  },
  menuItem: {
    display: "flex",
    [theme.breakpoints.up("sm")]: {
      display: "none",
    },
  },
  signInMobileMenu: {
    textDecoration: "none",
    color: "#0277bd",
  },
  signUpMobileMenu: {
    textDecoration: "none",
    color: "#0277bd",
  },
}));
