// components/BasketCart/index.js
import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import { connect } from 'react-redux';
import ShoppingCartIcon from '@material-ui/icons/ShoppingCart';

import { getTotalBasketCount, getTotalBasketPrice } from '../../selectors';

const useStyles = makeStyles({
  root: {
    textDecoration: "none",
    width: "100%",
    '& button': {
      textTransform: "none",
      height: 40,
    }
  }
});

const BasketCart = ({ totalBasketCount, totalBasketPrice }) => {
  const classes = useStyles();

  return (
    <Link to="/basket" className={classes.root}>
      <Button variant="contained" color="primary" fullWidth startIcon={<ShoppingCartIcon />}>
        {totalBasketCount} item(s) - $ {totalBasketPrice}
      </Button>
    </Link>
  );
};

const mapStateToProps = state => {
  return {
    totalBasketCount: getTotalBasketCount(state),
    totalBasketPrice: getTotalBasketPrice(state)
  }
}

export default connect(mapStateToProps)(BasketCart);
