// components/PhoneCard/index.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';

const useStyles = makeStyles({
  root: {
    maxWidth: 345,
  },
  title: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    '& span': {
      fontSize: "1rem",
      fontWeight: "bold",
    },
  },
  link: {
    color: "#007bff",
    textDecoration: "none",
    '&:hover': {
      textDecoration: "underline",
    }
  }
});

const PhoneCard = ({ phone }) => {
  // console.log("TCL: PhoneCard -> props", props)
  const classes = useStyles();
  const shortDescription = `${phone.description.slice(0, 61)}...`;

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia
          component="img"
          alt={phone.name}
          // height="140"
          image={phone.image}
          title={phone.name}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h3" className={classes.title}>
            <Link to={`/phones/${phone.id}`} className={classes.link}>
              {phone.name}
            </Link>
            <span>${phone.price}</span>
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {shortDescription}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="primary">
          Buy now!
        </Button>
        <Link to={`/phones/${phone.id}`} className={classes.link}>
          <Button>More Info</Button>
        </Link>
      </CardActions>
    </Card>
  );
};

export default PhoneCard
