// containers/Basket/styles.js
import { makeStyles } from "@material-ui/core";

export default makeStyles(theme => ({
  root: {
    marginTop: 80,
    [theme.breakpoints.up("sm")]: {
      marginTop: 90,
    },
    [theme.breakpoints.up("md")]: {
      marginTop: 105,
    },
    '& table tr:nth-of-type(even)': {
      background: "#eee",
    },
    '& td': {
      border: "1px solid #ccc",
      padding: "5px 16px",
    },
  },
  image: {
    width: "12%",
    '& img': {
      maxWidth: 100,
    },
  },
  deletePhone: {
    cursor: "pointer",
    '&:hover': {
      color: [theme.palette.error.main],
    },
  },
  totalPrice: {
    display: "flex",
    justifyContent: "flex-end",
    margin: "10px 0",
    '& b': {
      marginRight: 5,
    },
  },
  continueShopping: {
    display: "block",
    background: [theme.palette.info.main],
    textDecoration: "none",
    borderRadius: 5,
    marginBottom: 10,
    '&:hover': {
      background: [theme.palette.info.dark],
    },
    '& span': {
      color: "white",
    },
  },
  cleanBasket: {
    background: [theme.palette.error.main],
    color: "white",
    marginBottom: 10,
    '&:hover': {
      background: [theme.palette.error.dark],
    },
  },
  checkout: {
    background: [theme.palette.success.main],
    color: "white",
    '&:hover': {
      background: [theme.palette.success.dark],
    },
  },
}));
