// taskList/selectors.js

let maxId = 0

export const createItem = (text) => {
  return {
    label: text,
    important: false,
    done: false,
    id: maxId++
  }
}

export const toggleProperty = (arr, id, propName) => {
  // const newTasks = [...tasks]
  // const task = newTasks.find(task => task.id === id)
  // task.important = !task.important
  // setTasks(newTasks)

  const idx = arr.findIndex(item => item.id === id)
  const oldItem = arr[idx]
  // newItem[propName] = !newItem[propName]
  const newItem = {
    ...oldItem,
    [propName]: !oldItem[propName]
  }

  return [
    ...arr.slice(0, idx),
    newItem,
    ...arr.slice(idx + 1)
  ]
}

export const deleteItem = (arr, id) => {
  // const newArr = [...arr]
  // return newArr.filter(item => item.id !== id)

  const idx = arr.findIndex(item => item.id === id)
  return [
    ...arr.slice(0, idx),
    ...arr.slice(idx + 1)
  ]
}

export const search = (arr, term) => {
  if (term.trim() === '') {
    return arr
  }

  return arr.filter(item => item.label
    .toLowerCase()
    .indexOf(term.toLowerCase()) > -1
  )
}

export const filter = (arr, filter) => {
  switch (filter) {
    case 'all':
      return arr

    case 'active':
      return arr.filter(item => !item.done)

    case 'done':
      return arr.filter(item => item.done)

    default:
      return arr
  }
}
