// containers/Phone/styles.js
import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  root: {
    paddingLeft: 15,
    paddingRight: 15,
    marginTop: 80,
    [theme.breakpoints.up("sm")]: {
      marginTop: 95,
    },
    [theme.breakpoints.up("md")]: {
      marginTop: 105,
    },
  },
  info: {
    marginBottom: 30,
  },
  image: {
    width: "100%",
    height: "auto",
    maxWidth: 300,
  },
  field: {
    width: "100%",
    display: "flex",
    alignItems: "center",
    padding: "10px 0",
    borderBottom: "1px dotted #ccc",
    '& h6': {
      marginRight: 10,
    },
  },
  title: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "baseline",
    '& span': {
      fontSize: "1.5rem",
    },
  },
  titlePhone: {
    margin: "20px 0 0",
  },
  link: {
    display: "block",
    textDecoration: "none",
    marginBottom: 10,
  },
  backToStore: {
    color: "white",
    height: 40,
    backgroundColor: theme.palette.info.main,
    '&:hover': {
      backgroundColor: theme.palette.info.dark,
    },
  },
  addPhoneToBasket: {
    color: "white",
    height: 40,
    backgroundColor: theme.palette.success.main,
    '&:hover': {
      backgroundColor: theme.palette.success.dark,
    },
  },
}));
