// containers/auth/SignUp/index.js
import React, { useState } from 'react';
import { TextField, Typography, Paper, Button } from '@material-ui/core';
import { Link } from 'react-router-dom';
import DraftsTwoToneIcon from '@material-ui/icons/DraftsTwoTone';
import VpnKeyTwoToneIcon from '@material-ui/icons/VpnKeyTwoTone';
import LockOpenTwoToneIcon from '@material-ui/icons/LockOpenTwoTone';

import useStyles from 'containers/auth/useStyles';
import { validateControl } from 'selectors';

const SignUp = () => {
  const classes = useStyles()
  const [formControls, setFormControls] = useState({
    email: {
      type: 'email',
      label: 'Email',
      value: '',
      valid: false,
      touched: false,
      errorMessage: 'Enter correct email.',
      validation: {
        required: true,
        email: true,
      },
    },
    password: {
      type: 'password',
      label: 'Password',
      value: '',
      valid: false,
      touched: false,
      errorMessage: 'At least 6 characters.',
      validation: {
        required: true,
        minLength: 6,
      },
    },
    verifyPassword: {
      type: "password",
      label: "Verify Password",
      value: "",
      valid: false,
      touched: false,
      errorMessage: "Enter correct password.",
      validation: {
        required: true,
        samePassword: true,
      },
    },
  });

  const handleChange = controlName => event => {
    const newFormControls = { ...formControls };
    const control = newFormControls[controlName];

    control.value = event.target.value;
    control.touched = true;
    control.valid = validateControl(
      control.value,
      control.validation,
      formControls.password.value
    );

    setFormControls(newFormControls);
  };

  const header = (
    <div className={classes.header}>
      <Typography variant='h4' color='secondary'>
        Sign In
      </Typography>
      <Typography variant='h4' color='primary'>
        Sign Up
      </Typography>
    </div>
  );

  const renderFormControls = () => {
    return (
      Object.keys(formControls).map((controlName, index) => {
        const control = formControls[controlName];
        const isInvalid = !control.valid && control.touched;
        const helperText = isInvalid
          ? control.errorMessage || "Incorrect entry."
          : "";

        let icon = null;

        switch (controlName) {
          case 'email':
            icon = <DraftsTwoToneIcon fontSize="small" />;
            break;

          case 'password':
            icon = <VpnKeyTwoToneIcon fontSize="small" />;
            break;

          case 'verifyPassword':
            icon = <LockOpenTwoToneIcon fontSize="small" />;
            break;

          default:
            icon = null;
        }

        return (
          <div key={index} className={classes.controlField}>
            {icon}
            <TextField
              id={`${control.label}-${index}`}
              type={control.type}
              label={control.label}
              value={control.value}
              error={!control.valid && control.touched}
              helperText={helperText}
              className={classes.textField}
              onChange={handleChange(controlName)}
            />
          </div>
        );
      })
    );
  };

  return (
    <div className={classes.root}>
      <div className='auth'>
        <Paper elevation={10}>
          <form className={classes.form}>

            {header}

            {renderFormControls()}

            <Button variant="contained" color="primary" className={classes.button}>
              Sign In
            </Button>

            <Link to="sign-up" className={classes.privacyPolicy}>
              Privacy Policy
            </Link>

            <Link to="/" className={classes.cancel}>
              Cancel
            </Link>

          </form>
        </Paper>
      </div>
    </div>
  );
}

export default SignUp
