// internetShop/selectors.js
export const getPhoneById = (state, id) =>
  state.internetShop.phones.find(phone => phone.id === id)

export const getPhones = (state, { match }) => {
  const activeCategory = match.params.id || undefined
  const applySearch = phone => phone.name
    .toLowerCase()
    .indexOf(state.internetShop.phonesPage.search.toLowerCase()) > -1

  const applyCategory = phone => phone.categoryId === activeCategory

  const phones = state.internetShop.phonesPage.ids
    .map(id => getPhoneById(state, id))
    .filter(applySearch)

  if (activeCategory === undefined) {
    return phones
  }

  return phones.filter(applyCategory)
}

export const getTotalBasketCount = state => state.internetShop.basket.length

export const getTotalBasketPrice = state => {
  const applyTotalSum = (total, current) => total + current

  return state.internetShop.basket
    .map(id => getPhoneById(state, id))
    .map(phone => phone.price)
    .reduce(applyTotalSum, 0)
}

export const getCategories = state => {
  return state.internetShop.categories
}

export const getBasketPhonesWithCount = state => {
  const uniqueIds = [...new Set(state.internetShop.basket)]
  const phoneCount = id => state.internetShop.basket
    .filter(basketId => basketId === id)
    .length

  return uniqueIds
    .map(id => getPhoneById(state, id))
    .map(phone => ({...phone, count: phoneCount(phone.id)}))
}

export const validateEmail = email => {
  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  return re.test(email.toString().toLowerCase())
}

export const validateControl = (value, validation, newFormControls) => {
  if (!validation) {
    return true
  }

  let isValid = true

  if (validation.required) {
    isValid = value.trim() !== '' && isValid
  }

  if (validation.email) {
    isValid = validateEmail(value) && isValid
  }

  if (validation.minLength) {
    isValid = value.length >= validation.minLength && isValid
  }

  if (validation.samePassword) {
    // console.log("TCL: validateControl -> value", value)
    isValid = newFormControls.password.value === value && isValid
  }

  return isValid
}
